server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileResource.java

```diff
@@ -179,6 +179,14 @@ public TsFileResource(File file) {
     this.timeIndex = CONFIG.getTimeIndexLevel().getTimeIndex();
   }
 
+  /** Used for compaction to create target files. */
+  public TsFileResource(File file, TsFileResourceStatus status) {
+    this.file = file;
```

Comment:
using `this(file)` to reduce duplicated codes

Reply:
Resolved.